### PR TITLE
README.md: Clearly state which buckets to create beforehand as bootstrapping and which will be created by terraform

### PR DESCRIPTION
**Pull request by brainstorm**
Otherwise `make deploy-infra` will fail later on.

### CODE REVIEW

**Comment by brainstorm** [comment]
But the very previous README step is essentially inconsistent since the `make deploy-infra` terraform templates seem to (re-)create most of the buckets anyway:

```
Terraform will perform the following actions:

  + aws_s3_bucket.dss_s3_bucket
  + aws_s3_bucket.dss_s3_bucket_test
  + aws_s3_bucket.dss_s3_bucket_test_fixtures
  + aws_s3_bucket.dss_s3_checkout_bucket
  + aws_s3_bucket.dss_s3_checkout_bucket_test
  + aws_s3_bucket.dss_s3_checkout_bucket_unwritable

  + google_storage_bucket.dss_gs_bucket
  + google_storage_bucket.dss_gs_bucket_test
  + google_storage_bucket.dss_gs_bucket_test_fixtures
  + google_storage_bucket.dss_gs_checkout_bucket
  + google_storage_bucket.dss_gs_checkout_bucket_test
```

Which will most likely fail since those are created already. Therefore the whole [AWS S3 bucket creation README section should go away from the README.md](https://github.com/HumanCellAtlas/data-store#configure-aws)? I guess that the equivalent [section 12 of the GCP bucket creation](https://github.com/HumanCellAtlas/data-store#configure-gcp) should go as well?

**Comment by brainstorm** [comment]
And as expected, on a bunch of buckets, terraform fails to create them because they were already previously created in the README:

```
* aws_s3_bucket.dss_s3_bucket: Error creating S3 bucket: BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.
```

Causing the rest of the Makefile target fail:

```
Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


make[2]: *** [Makefile:37: apply] Error 1
make[2]: Leaving directory '/home/ubuntu/data-store/infra'
make[1]: *** [Makefile:17: apply-all] Error 1
make[1]: Leaving directory '/home/ubuntu/data-store/infra'
make: *** [Makefile:80: deploy-infra] Error 2
```

On a similar note, the GS buckets also fail to be created but in this case for insufficient permissions (followed the README to define them, but clearly this should be better automated instead):

```
* google_storage_bucket.dss_gs_checkout_bucket: googleapi: Error 403: umccr-org-hca-store@hca-data-store-XXXX.iam.gserviceaccount.com does not have storage.buckets.create access to project XXXXXXXXX., forbidden
```

File: README.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ AWS command line utility and configure your AWS access credentials.
 1. If you wish to run the unit tests, you must create two more S3 buckets, one for test data and another for test
    fixtures, and set the environment variables `DSS_S3_BUCKET_TEST` and `DSS_S3_BUCKET_TEST_FIXTURES` to the names of
    those buckets.
+   
+1. Also make sure to create the terraform S3 state bucket defined by `DSS_TERRAFORM_BACKEND_BUCKET_TEMPLATE`, this is crucial for deploying data-store later on.
```
Comment (xbrianh) [inline README.md]: We prefer the phrasing "The environment variable `DSS_TERRAFORM_BACKEND_BUCKET_TEMPLATE` must point at an S3 bucket before executing `make infra`. If this bucket does not already exist, it must be created."

**Comment by xbrianh** [comment]
@brainstorm Please see if these changes sufficiently address your concerns:  https://github.com/HumanCellAtlas/data-store/pull/1747

File: README.md (hunk 1 above)
Comment (brainstorm) [reply README.md]:
```suggestion
1. DSS_TERRAFORM_BACKEND_BUCKET_TEMPLATE must point at an S3 bucket before executing make infra. If this bucket does not already exist, it must be created.
```

**Comment by xbrianh** [comment]
resolved via #1924

**Pull request closed without merging**